@graydon looks like a legitimate test failure:

~~~
16:26:49 
/Users/buildnode/jenkins/workspace/swift-PR-osx/branch-swift-3.1-branch/swift/test/Misc/expression_too_complex_2.swift:12:24: error: unexpected error produced: expression was too complex to be solved in reasonable time; consider breaking up the expression into distinct sub-expressions
16:26:49     NSLayoutConstraint.activate([
16:26:49                        ^
16:26:49 
16:26:49 --
16:26:49 
16:26:49 ********************
16:26:49 PASS: Swift(iphonesimulator-i386) :: radar/r17963339.swift (865 of 9217)
16:26:49 PASS: Swift(iphonesimulator-i386) :: stdlib/CollectionDiagnostics.swift (866 of 9217)
16:26:50 PASS: Swift(iphonesimulator-i386) :: execution/testability.swift (867 of 9217)
16:26:51 PASS: Swift(iphonesimulator-i386) :: Constraints/dynamic_lookup.swift (868 of 9217)
~~~

--- a/test/Misc/expression_too_complex_2.swift
+++ b/test/Misc/expression_too_complex_2.swift
@@ -9,7 +9,7 @@ class MyViewCell: UITableViewCell {
   required init?(coder aDecoder: NSCoder) { fatalError("no") }
   override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
     super.init(style: .default, reuseIdentifier: reuseIdentifier)
-    NSLayoutConstraint.activate([
+    NSLayoutConstraint.activate([ // expected-error{{expression was too complex to be solved in reasonable time; consider breaking up the expression into distinct sub-expressions}}
         NSLayoutConstraint(item: View1, attribute: .top, relatedBy: .equal, toItem: label1, attribute: .top, multiplier: 1, constant: 1),
         NSLayoutConstraint(item: View1, attribute: .top, relatedBy: .equal, toItem: label2, attribute: .top, multiplier: 1, constant: 1),
         NSLayoutConstraint(item: View1, attribute: .top, relatedBy: .equal, toItem: label3, attribute: .top, multiplier: 1, constant: 1),

